src/util/xml.cpp

```diff
@@ -34,8 +34,7 @@ void xmlt::output(std::ostream &out, unsigned indent) const
   // 'name' needs to be set, or we produce mal-formed
   // XML.
 
-  if(name.empty())
-    return;
+  PRECONDITION(!name.empty());
```

Comment:
That'd make `xmlt` not default constructible anymore which'd be a bigger change.

I'm also noticing that the `xmlt` class seems to be limited to a single piece of text content, rather than allowing text and markup to intersperse like normal xml, but that's also a different issue.